### ISSUE TITLE
[Question] remote harvester config: x509: certificate signed by unknown authority

### ISSUE THREAD

**Issue by gsuess** [harvester/harvester#4134]
On the ISO installation, when I attempt to load a remote harvester config, I get the issue that the self-signed certificate is not trusted:

![rpviewer (2)](https://github.com/harvester/harvester/assets/1558575/004df64f-5aaa-49b9-aa09-b8956abe3503)

I have tried to trust the crt on another tty:

```bash
sudo trust anchor certificate.crt
```

But that somehow didn't have any effect.

**Comment by w13915984028** [comment]
@gsuess   please try use HTTP instead of HTTPS here, thanks.

https://docs.harvesterhci.io/dev/install/iso-install

`13 (Optional) If you need to customize the host with a [Harvester configuration](https://docs.harvesterhci.io/dev/install/harvester-configuration) file, enter the HTTP URL here.`

![image](https://github.com/harvester/harvester/assets/31133476/ecc53ede-2641-473c-8f22-0ff4c30bc41a)

**Comment by Martin-Weiss** [comment]
@w13915984028  - I believe we really need support for https as well. Many environments do not allow port 80 to be used / opened in firewalls, anymore. Not sure if it would help to use ssl.certs=0 on the kernel cmdline initially to workaround the above problem? (not sure if the installer supports this, already)

**Comment by gsuess** [comment]
I ended up working around it by pasting it into a secret gist and then using its https endpoint which is trusted by default.

I still wonder why adding the crt using `trust` didn't work though.

**Comment by devclef** [comment]
I've had this issue on fresh installs on additional nodes when using https:// for the harvester URL, using let's encrypt certs for harvester. 

After the restart the node fails to join cluster with above error. I've solved this by placing the certs into /etc/pki/trust/anchors/ post-install,  the node then joins the cluster without issue.

Not sure if different for fetching the harvester configuration file as OP, but perhaps worth trying as a work around?

**Comment by tserong** [comment]
The installer is just using go's `http.Get()` (see https://github.com/harvester/harvester-installer/blob/c3aab22453111b4391136a9740a1db115f57fc8f/pkg/console/util.go#L69), so to fix this properly we'd need to either add a flag somewhere to allow disabling SSL certificate verification (i.e. use `tls.Config.InsecureSkipVerify`), or add some means of injecting additional CA certs into the installer.

I don't yet know why `sudo trust anchor certificate.crt` didn't help, I'll have to investigate that a bit further.

**Comment by tserong** [comment]
I tried `sudo trust anchor certificate.crt`, with a self-signed certificate. Turns out there's two problems with this:

1. `trust anchor` adds the certificate as "category: other-entry", which doesn't work. It needs to be "category: authority" for the validation to happen (check the output of `trust anchor list` and compare the categories). It seems you can kinda work around this by editing `/etc/pki/trust/$YOUR_CERT.p11-kit` and changing `certificate-category: other-entry` to `certificate-category: authority`, but I strongly doubt this is recommended practice :-)
2. Once you get the above sorted out, `harvester-installer` won't re-read the CA certs on the fly - it needs to be restarted to pick up those changes. It's possible to do this by running `systemctl restart getty@tty1.service`.

TL;DR: the above might work, but I do _not_ recommend it. We really need a proper fix for handling self-signed certificates.

**Comment by Yu-Jack** [comment]
@tserong It seems this issue has been clarified. I think we can put it into development, right?
